src/overlays/actors/ovl_Obj_Oshihiki/z_obj_oshihiki.c

```diff
@@ -27,63 +67,661 @@ const ActorInit Obj_Oshihiki_InitVars = {
+                if ((dynaSwitchFlag == ((this->dyna.actor.params >> 8) & 0x3F)) &&
+                    Flags_GetSwitch(globalCtx, dynaSwitchFlag)) {
+                    return 0;
+                }
+                break;
+            case 0x30: // (OBJSWITCH_SUBTYPE_FLOOR_3 << 4) | OBJSWITCH_TYPE_FLOOR
+                if ((dynaSwitchFlag == ((this->dyna.actor.params >> 8) & 0x3F)) &&
+                    !Flags_GetSwitch(globalCtx, dynaSwitchFlag)) {
+                    return 0;
+                }
+                break;
+        }
+    }
+    return 1;
+}
+
+void ObjOshihiki_CheckType(ObjOshihiki* this, GlobalContext* globalCtx) {
+    switch (BLOCK_TYPE) {
+        case SMALL_SWITCH_OFF:
+        case MEDIUM_SWITCH_OFF:
+        case LARGE_SWITCH_OFF:
+        case HUGE_SWITCH_OFF:
+        case SMALL_SWITCH_ON:
+        case MEDIUM_SWITCH_ON:
+        case LARGE_SWITCH_ON:
+        case HUGE_SWITCH_ON:
+            ObjOshihiki_InitDynapoly(this, globalCtx, &D_05004E98, 1);
+            return;
+    }
+    // Error : type cannot be determined
+    osSyncPrintf("Error : タイプが判別できない(%s %d)(arg_data 0x%04x)\n", "../z_obj_oshihiki.c", 444,
+                 this->dyna.actor.params);
+}
+
+void ObjOshihiki_SetScale(ObjOshihiki* this, GlobalContext* globalCtx) {
+    Actor_SetScale(&this->dyna.actor, sScales[BLOCK_TYPE]);
+}
+
+void ObjOshihiki_SetTexture(ObjOshihiki* this, GlobalContext* globalCtx) {
+    switch (BLOCK_TYPE) {
+        case SMALL_SWITCH_OFF:
+        case MEDIUM_SWITCH_OFF:
+        case SMALL_SWITCH_ON:
+        case MEDIUM_SWITCH_ON:
+            this->texture = &D_05003350;
+            break;
+        case LARGE_SWITCH_OFF:
+        case LARGE_SWITCH_ON:
+            this->texture = &D_05003B50;
+            break;
+        case HUGE_SWITCH_OFF:
+        case HUGE_SWITCH_ON:
+            this->texture = &D_05004350;
+            break;
+    }
+}
+
+void ObjOshihiki_SetColor(ObjOshihiki* this, GlobalContext* globalCtx) {
+    Color_RGB8* src;
+    Color_RGB8* color = &this->color;
+    s16 paramsColorIdx;
+    s32 i;
+
+    paramsColorIdx = (this->dyna.actor.params >> 6) & 3;
+
+    for (i = 0; i < ARRAY_COUNT(sScenes); i++) {
+        if (sScenes[i] == globalCtx->sceneNum) {
+            break;
+        }
+    }
+
+    if (i > 8) {
+        // "Error : scene_data_ID cannot be determined"
+        osSyncPrintf("Error : scene_data_ID が判別できない。(%s %d)\n", "../z_obj_oshihiki.c", 579);
+        color->r = color->g = color->b = 255;
+    } else {
+        src = &sColors[i][paramsColorIdx];
+        color->r = src->r;
+        color->g = src->g;
+        color->b = src->b;
+    }
+}
+
+void ObjOshihiki_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjOshihiki* this = THIS;
+    s32 switchFlag;
+
+    ObjOshihiki_CheckType(this, globalCtx);
+
+    switchFlag = this->dyna.actor.params >> 8;
+
+    if (((u8)switchFlag >= 0) && ((u8)switchFlag <= 0x3F)) {
```

Comment:
Any chance switchFlag could be a `u8`? or the `& 0xFF` is part of the line above where it extracts it from params?

Reply:
The temp turns out to be unnecessary, but the (u8) casts are.

Reply:
The u8 casts are very likely fake since it should be `& 0xFF` instead, so I would try to get rid of them however you can

Reply:
I'll see what I can do, but it may end up being `((switchFlag & 0xFF) >= 0) && ((switchFlag & 0xFF) <= 0x3F)`

Reply:
I think the best I'm doing is `& 0xFF` without a temp. There was no nice looking way to use a temp and have the regalloc work. Also I'm pretty sure no temp is used for the flag in the switch check because it's clearly a macro.